src/main/java/bio/terra/service/dataset/StorageResource.java

```diff
@@ -2,39 +2,74 @@
 
 import bio.terra.app.model.CloudRegion;
 import bio.terra.app.model.CloudResource;
+import bio.terra.app.model.GoogleCloudResource;
+import bio.terra.app.model.GoogleRegion;
 import bio.terra.model.CloudPlatform;
 import bio.terra.model.StorageResourceModel;
 import com.fasterxml.jackson.annotation.JsonSubTypes;
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
 
+import java.util.Objects;
 import java.util.UUID;
 
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXISTING_PROPERTY, property = "cloudPlatform")
 @JsonSubTypes({
     @JsonSubTypes.Type(value = AzureStorageResource.class, name = "azure"),
     @JsonSubTypes.Type(value = GoogleStorageResource.class, name = "gcp")
 })
-public interface StorageResource<Region extends CloudRegion, Resource extends CloudResource> {
+public abstract class StorageResource<Region extends CloudRegion, Resource extends CloudResource> {
```

Comment:
I thought you were trying to do away with the generics?

Reply:
I think I'm close to getting a version without the generics to work

Reply:
Well, I was trying it out and it didn't seem so bad. I like that the subclasses are pretty much marker classes with this change.

Reply:
Welp, nevermind...couldn't get it to work without the generics...No a huge deal, just means that we need a bunch of `<?, ?>`'s in the code when we have `StorageResource` variables